server/util/muster-utils.ts

```diff
@@ -92,67 +92,74 @@ export async function getRosterMusterStats(args: {
       .getMany();
   }
 
+  const rosterStats: MusterRosterStats = {};
+
   const rosterEntriesByEdipi: { [edipi: string]: Roster } = {};
-  rosterEntries.forEach(e => {
-    rosterEntriesByEdipi[e.edipi] = e;
+  rosterEntries.forEach(entry => {
+    const { edipi } = entry;
+
+    rosterEntriesByEdipi[edipi] = entry;
+
+    if (!rosterStats[edipi]) {
+      rosterStats[edipi] = {
+        totalMusters: 0,
+        mustersReported: 0,
+        musterPercent: 0,
+      };
+    }
 
     // Fill in any missing phone numbers that weren't in ES.
     // HACK: Just try to find any custom column with "phone" in its name. Maybe we need a special type
     // for phone number, or need to make it a non-custom column...
-    const phoneKey = Object.keys(e.customColumns)
+    const phoneKey = Object.keys(entry.customColumns)
       .find(key => key.toLowerCase().indexOf('phone') !== -1);
 
-    if (phoneKey && edipiToPhone[e.edipi] == null) {
-      edipiToPhone[e.edipi] = e.customColumns[phoneKey] as string;
+    if (phoneKey && edipiToPhone[entry.edipi] == null) {
+      edipiToPhone[entry.edipi] = entry.customColumns[phoneKey] as string;
     }
   });
 
   const allowedRosterColumns = await Roster.getAllowedColumns(org, userRole.role);
 
   // Collect reports and reports missed.
-  const individualStats: IndividualStats = {};
-
-  const aggs = musterAggsResponse.aggregations as IndividualsMusterAggregations | undefined;
+  const aggs = musterAggsResponse.aggregations as MusterRosterAggregations | undefined;
   const buckets = aggs?.muster.buckets ?? [];
   for (const bucket of buckets) {
     const { edipi, reported } = bucket.key;
 
-    if (!individualStats[edipi]) {
-      individualStats[edipi] = {
+    if (!rosterStats[edipi]) {
+      rosterStats[edipi] = {
+        totalMusters: 0,
         mustersReported: 0,
-        mustersNotReported: 0,
-        nonMusterPercent: 0,
+        musterPercent: 0,
       };
     }
 
+    rosterStats[edipi].totalMusters += bucket.doc_count;
+
     if (reported) {
-      individualStats[edipi].mustersReported = bucket.doc_count;
-    } else {
-      individualStats[edipi].mustersNotReported = bucket.doc_count;
+      rosterStats[edipi].mustersReported = bucket.doc_count;
     }
   }
 
-  // Calculate non-muster percents.
-  for (const edipi of Object.keys(individualStats)) {
-    const data = individualStats[edipi];
-    individualStats[edipi].nonMusterPercent = calcNonMusterPercent(data.mustersReported, data.mustersNotReported);
+  // Calculate muster percents.
+  for (const edipi of Object.keys(rosterStats)) {
```

Comment:
Might be cleaner as `for (const [edipi, data] = Object.entries(rosterStats)) {`

Reply:
Looks like it can even be `Object.values()` and just skip the key altogether.